pkg/kubelet/dockertools/exec.go

```diff
@@ -96,10 +94,16 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		err = command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
 	}
-
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
```

Comment:
This timeout func is deferred until the function returns and `command.Wait()` has long been executed. You want:

```go
t :=  time.AfterFunc(timeout, func() { ... })
defer t.Stop()
```
don't you?